### PR TITLE
Be able to parse key-direction in our configuration file

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

**Comment by dinosaure** [comment]
Fix #207

**Review by reynir — COMMENTED** [review]
I'm curious if this is a valid configuration:
```
key-direction 1
tls-auth [inline]
<tls-auth>
...
</tls-auth>
```
And also with `key-direction` and `tls-auth [inline]` swapped.

other than that I think it looks fine.

**Comment by dinosaure** [comment]
Actually, if you put `tls-auth [inline]`, it leads to an error because we expect 2 inline parameters. Not sure how to fix it and if we should do.

**Review by reynir — APPROVED** [review]
It improves the situation even if there is an unhandled corner case. If you could `dune build @fmt` then I think it's good to merge.

File: test/config_tests.ml [hunk 1]
```diff
@@ -381,6 +381,7 @@ let key_direction () =
     Fmt.str
       {|%a
 key-direction 1
+tls-auth [inline]
```
Comment (reynir) [inline test/config_tests.ml:384]: Did you mean to commit this line? As I understand your previous comment this makes the test fail.

Comment (dinosaure) [reply test/config_tests.ml]: Ah yes, I did the test locally and forget to remove it, sorry.
